example_plugins/foo/src/index.ts

```diff
@@ -1,16 +1,27 @@
 import { KibanaPluginConfig } from '@elastic/kbn-types';
+import { k$, first, toPromise } from '@elastic/kbn-observable';
 import { BarPluginType } from 'example-plugin-bar';
+import { FooConfig } from './FooConfig';
 
 export const plugin: KibanaPluginConfig<BarPluginType> = {
-  configPath: ['foo'],
+  configPath: ['__newPlatform', 'foo'],
   dependencies: ['bar'],
-  plugin: (kibana, dependencies) => {
+  async plugin(kibana, dependencies) {
```

Comment:
Not related to this PR, but something we've got to decide: do we `await` when starting up every plugin? Right now they are started synchronously and in topological order, so `await`-ing means later plugins must wait for this to finish. I don't see any problem with that (and I expect we want an async `stop` too).

There's a todo in https://github.com/elastic/kibana/blob/new-platform/platform/server/plugins/Plugin.ts#L99.

(Let's not do anything for this PR, but if you want to look at that after, it would be 🎉)

Reply:
Yep, good point. I'll put this into my own roadmap :)